backend/tracim_backend/__init__.py

```diff
@@ -333,6 +335,8 @@ def adapt_auth_policy(authentication_policy) -> SecurityPolicyAdaptor:
     plugin_manager.hook.web_include(configurator=configurator, app_config=app_config)
 
     hapic.add_documentation_view("/api/doc", "Tracim API", "API of Tracim")
+
+    configurator.end()
```

Comment:
This pops the `Registry` from the `threadlocal` stack, propagating the changes.

Reply:
> propagating the changes.

To the entire app? If so, I'm afraid the complete system does not work:

- the configuration would be updated in each thread (threadlocal means the execution is local for a thread, not executed in a unique thread) then applied N times according to the number of threads?

I don't know how pyramid is designed for this stuff, I'm ok to merge the PR *in case you are sure of the behavior you expect* and that *it works like you/we imagine*.

Reply:
This behavior is done internally when providing the `autocommit=True` parameter to the constructor of the `Configurator`. Putting it manually is manually triggering this propagation since we manually popped the `Configurator` to the stack.

To make sure of what I say, I tested it locally. Changes done from the plugins or read from the config file are indeed applied to the app.

I am sure of the behavior and it works as I imagined it.

Reply:
Ok I understand. To make sure I tested with verifiable events and 4 threads, whether it is prints, requests to external elements etc.. It is only ran once. I tested with `uwsgi` and the following settings:

```
workers = 4
threads = 4
```

I have not tested SQL yet, but it seems configuration is resolved before threading anything or starting processes. I have confirmation that configuration code (and so plugins configuration) is only ran once no matter the numbers of workers/threads

Reply:
Nice work @Millefeuille42 . So I suggest to add this information to the documentation - most of developers won't ask for it but the ones who will will also have answer to their questions

Reply:
I added this information in the plugins doc in the "technical information" section. It is easier to find it here, since plugins dev *should* read the documentation (at least more that issues).